src/plugins/ramps/paybis/paybisRampPlugin.ts

```diff
@@ -0,0 +1,1360 @@
+          if (
+            pair.from === fiat &&
+            pair.to.some(to => to.currencyCode === paybisCc)
+          ) {
+            pairSupported = true
+            break
+          }
+        }
+      } else {
+        const sellPairs = paymentMethodPairs as ReturnType<
+          typeof asSellPaymentMethodPairs
+        >
+        for (const pair of sellPairs.pairs) {
+          if (pair.fromAssetId === paybisCc && pair.to.includes(fiat)) {
+            pairSupported = true
+            break
+          }
+        }
+      }
+      if (pairSupported) break
+    }
+
+    if (!pairSupported) {
+      return { supported: false }
+    }
+
+    return undefined
+  }
+
+  const plugin: RampPlugin = {
+    pluginId,
+    rampInfo,
+
+    checkSupport: async (
+      request: RampCheckSupportRequest
+    ): Promise<RampSupportResult> => {
+      try {
+        const { direction, regionCode, fiatAsset, cryptoAsset } = request
+
+        // Ensure assets are initialized for the direction
+        await ensureAssetsInitialized(direction)
+
+        // Validate region and country restrictions
+        const regionResult = validateSupportRequest(
+          regionCode,
+          regionCode.countryCode
+        )
+        if (regionResult != null) {
+          return regionResult
+        }
+
+        // Check asset support
+        const assetResult = checkAssetSupport(
+          direction,
+          `iso:${fiatAsset.currencyCode}`,
+          cryptoAsset.pluginId,
+          cryptoAsset.tokenId
+        )
+        if (assetResult != null) {
+          return assetResult
+        }
+
+        // If we get here, it's supported
+        return { supported: true }
+      } catch (error) {
+        // Only throw for actual errors (network issues, etc)
+        // Never throw for unsupported combinations
+        console.error('Paybis checkSupport error:', error)
+        throw error
+      }
+    },
+
+    fetchQuote: async (
+      request: RampQuoteRequest
+    ): Promise<RampQuoteResult[]> => {
+      await ensureStateInitialized()
+      if (!state) throw new Error('Plugin state not initialized')
+
+      const {
+        amountType,
+        exchangeAmount,
+        regionCode,
+        pluginId: currencyPluginId,
+        promoCode: maybePromoCode,
+        pluginUtils,
+        fiatCurrencyCode,
+        displayCurrencyCode,
+        direction,
+        tokenId
+      } = request
+
+      // Validate region and country restrictions using helper
+      const regionResult = validateSupportRequest(
+        regionCode,
+        regionCode.countryCode
+      )
+      if (regionResult != null) {
+        // Return empty array for unsupported regions
+        return []
+      }
+
+      // Initialize assets for the direction
+      await ensureAssetsInitialized(direction)
+
+      // Get all supported payment types for the direction
+      let allPaymentTypes = Object.keys(allowedPaymentTypes[direction]).filter(
```

Comment:
If there is a way to know which payments types are valid for a specific country, that would save you having to get quotes for all payment types. This might have to be hard coded in each plugin if they don't have an API for it.

Reply:
So what @paullinator is saying is that we define which payment types are valid on a per-plugin basis. Though, we don't filter payment types by the region parameter within `fetchQuote` routine when we could. Because payment types are semantically region specific, this would make senses to do and optimize for. Until we create a map of payment types and their region or region to payment type, we can leave this suggestion unchanged.